src/taskgraph/morph.py

```diff
@@ -165,12 +166,22 @@ def add_index_tasks(taskgraph, label_to_taskid, parameters, graph_config):
     """
     logger.debug("Morphing: adding index tasks")
 
+    task_index_config = graph_config["taskgraph"].get("task-index", {})
     added = []
     for label, task in taskgraph.tasks.items():
-        if len(task.task.get("routes", [])) <= MAX_ROUTES:
+        if (
+            not task_index_config.get("all", False)
+            and task.kind not in task_index_config.get("kinds", [])
+            and task.label not in task_index_config.get("labels", [])
+            and len(task.task.get("routes", [])) <= MAX_ROUTES
```

Comment:
This "index-task" is only needed if we go over the maximum routes allowed for a task. Most tasks don't go over this maximum. So the "eager" solution will need to work for the case we don't have an index task as well (or does it already work for that case?)